hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt

```diff
@@ -60,15 +62,23 @@ class HmppsQueueFactory(
         createDefaultBean().also { bean -> context.beanFactory.registerSingleton(beanName, bean) }
       }
 
-  private fun createSqsListenerContainerFactory(hmppsQueue: HmppsQueue): HmppsQueueDestinationContainerFactory =
+  private fun createSqsListenerContainerFactory(hmppsQueue: HmppsQueue, errorVisibilityTimeout: Int): HmppsQueueDestinationContainerFactory =
     getOrDefaultBean("${hmppsQueue.id}-sqs-listener-factory") {
-      HmppsQueueDestinationContainerFactory(hmppsQueue.id, createSqsListenerContainerFactory(hmppsQueue.sqsClient))
+      HmppsQueueDestinationContainerFactory(hmppsQueue.id, createSqsListenerContainerFactory(hmppsQueue.sqsClient, errorVisibilityTimeout))
     }
 
-  private fun createSqsListenerContainerFactory(awsSqsClient: SqsAsyncClient): SqsMessageListenerContainerFactory<Any> =
+  private fun createSqsListenerContainerFactory(awsSqsClient: SqsAsyncClient, errorVisibilityTimeout: Int): SqsMessageListenerContainerFactory<Any> =
     SqsMessageListenerContainerFactory
       .builder<Any>()
       .sqsAsyncClient(awsSqsClient)
+      .errorHandler(object : ErrorHandler<Any> {
+        override fun handle(message: org.springframework.messaging.Message<Any>, t: Throwable) {
+          // SDI-477 remove this logging when we are comfortable that all is working as expected - instant retries
+          log.info("Setting visibility of messageId ${message.headers["id"]} to $errorVisibilityTimeout (to initiate faster retry) after receiving exception ${t.cause?.cause?.cause}")
+          val sqsVisibility = message.headers["Sqs_Visibility"] as QueueMessageVisibility
+          sqsVisibility.changeTo(errorVisibilityTimeout)
+        }
+      })
```

Comment:
The main change to timeout messages in error immediately (errorVisibilityTimeout defaults to 0).